toml/src/main/kotlin/org/rust/toml/inspections/CrateVersionInvalidInspection.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.toml.inspections
+
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.PsiElementVisitor
+import org.rust.toml.crates.local.CrateVersion
+import org.rust.toml.crates.local.CrateVersionRequirement
+import org.rust.toml.crates.local.CratesLocalIndexService
+import org.rust.toml.stringValue
+
+class CrateVersionInvalidInspection : CargoTomlInspectionToolBase() {
+    override val requiresLocalCrateIndex = true
+
+    override fun buildCargoTomlVisitor(holder: ProblemsHolder): PsiElementVisitor {
+        return object : CargoDependencyCrateVisitor() {
+            override fun visitDependency(dependency: DependencyCrate) {
+                if (dependency.isForeign()) return
+
+                val service = CratesLocalIndexService.getInstance()
+                val crate = service.getCrate(dependency.crateName) ?: return
```

Comment:
On `master`, `CratesLocalIndexService::getCrate(crateName: String)` now can throw `CratesLocalIndexException` in cases when index is not available when it either being updated or not available at all. Let's ignore inspections when the exception is caught